packages/storybook/stories/MessageThread/snippets/MessageWithCustomAttachment.snippet.tsx

```diff
@@ -0,0 +1,103 @@
+import {
+  AttachmentMenuAction,
+  AttachmentMetadata,
+  ChatMessage,
+  FluentThemeProvider,
+  MessageStatus,
+  MessageThread,
+  defaultAttachmentMenuAction
+} from '@azure/communication-react';
+import { initializeFileTypeIcons } from '@fluentui/react-file-type-icons';
+import { ShareIos24Regular, WindowNew24Regular } from '@fluentui/react-icons';
+import React from 'react';
+
+initializeFileTypeIcons();
+
+export const MessageWithCustomAttachment: () => JSX.Element = () => {
+  return (
+    <FluentThemeProvider>
+      <MessageThread
+        userId={'1'}
+        showMessageDate={true}
+        attachmentOptions={{
+          downloadOptions: {
+            actionsForAttachment: (attachment: AttachmentMetadata, message?: ChatMessage): AttachmentMenuAction[] => {
+              // allows the menu actions to be dynamically generated based on the attachment content
+              // (i.e. name, extension, etc.), or the message content (i.e. senderID, etc.)
+              // in this example, we are constructing a dynamic menu based on the attachment extension
+              if (attachment.extension === 'pdf') {
+                return [
+                  {
+                    name: 'Share',
+                    icon: <ShareIos24Regular />,
+                    onClick: () => {
+                      return new Promise((resolve) => {
+                        window.open(attachment.url, '_blank');
```

Comment:
Recommend just doing an alert here. People testing / running the sample may click it and be navigated to a url that doesn't exist. We do the same thing with the image overlay samples, when users tap the download button.